src/typhoon_automator/automator.py

```diff
@@ -151,7 +154,9 @@ def add_data_logger_signals(
         :param list[str] signals: Names of streaming signals to be logged
         :raises ValueError: The loaded schematic does not contain one or more of the signal names
         """
-        raise NotImplementedError()
+        if not signals or len(signals) == 0:
+            raise ValueError('The loaded schematic does not contain one or more of the signal names')
+        self._data_log_signals = signals[:]
```

Comment:
This method should add signals to the list, not replace the list:
```suggestion
        self._data_log_signals.append(signals)
```